addons/website_forum/controllers/main.py

```diff
@@ -164,23 +172,40 @@ def tag_read(self, q='', l=25, **post):
         )
         return simplejson.dumps(data)
 
-    @http.route(['/forum/<model("forum.forum"):forum>/tag'], type='http', auth="public", website=True)
-    def tags(self, forum, page=1, **post):
-        tags = request.env['forum.tag'].search([('forum_id', '=', forum.id), ('posts_count', '>', 0)], limit=None, order='posts_count DESC')
+    @http.route(['/forum/<model("forum.forum"):forum>/tag','/forum/<model("forum.forum"):forum>/tag/<tag_char>'], type='http', auth="public", website=True)
+    def tags(self, forum, page=1, tag_char=False, **post):
+        pager_chars = forum.get_tags_char()
+        tag_char = urllib.unquote(tag_char).decode("unicode_escape") if tag_char else (pager_chars[0][0] if pager_chars else 'All')
```

Comment:
It is because of non-ascii characters.Like Chinese, Hindi, Latin etc
And that's why i need to use decoding and why i can not use <string(length=1)
![download](https://cloud.githubusercontent.com/assets/7629089/5933794/7c37f97a-a6eb-11e4-886a-256bd795cfb2.png)